hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/ObjectEndpoint.java

```diff
@@ -245,6 +257,8 @@ public Response put(
         throw newError(S3ErrorTable.ACCESS_DENIED, keyPath, ex);
       } else if (ex.getResult() == ResultCodes.BUCKET_NOT_FOUND) {
         throw newError(S3ErrorTable.NO_SUCH_BUCKET, bucketName, ex);
+      } else if (ResultCodes.FILE_ALREADY_EXISTS == ex.getResult()) {
+        throw newError(S3ErrorTable.INVALID_REQUEST, keyPath, ex);
```

Comment:
I think this case warrants its own entry in the `S3ErrorTable` so that the end user with the s3 client sees a message explaining what happened. This will be very confusing otherwise.